### PR TITLE
WIP documentation

### PR DESCRIPTION
**Pull request by kaatish**
Documentation for functions.h

### CODE REVIEW

File: cpp/include/functions.h [hunk 1]
```diff
@@ -15,65 +15,189 @@
...
                              const gdf_column *offsets,
                              const gdf_column *indices,
                              const gdf_column *edge_data);
 
+/**
+ * @Synopsis   Create the adjacency lists of a gdf_graph from its edge list.
+ *             cuGRAPH allocates and owns the memory required for storing the created adjacency list.
+ *             The call expects a valid structure pointed by graph->edgeList. Upon successful completion of the call graph->adjList will point to a gdf_adj_list structure containing the generated adjacency list.
+ *
+ * @Param[out] *graph                cuGRAPH graph descriptor containing the newly added adjacency list.
```
Comment (afender) [inline cpp/include/functions.h]: `graph` is a `param[in, out]`. I would recommend adding the paragraph just above it to explain the [in] part.

File: cpp/include/functions.h [hunk 2]
```diff
@@ -15,65 +15,189 @@
...
+ */
+/* ----------------------------------------------------------------------------*/
 gdf_error gdf_add_adj_list(gdf_graph *graph);
+
+/**
+ * @Synopsis   Create the transposed adjacency list from the edge list of a gdf_graph.
+ *             cuGRAPH allocates and owns the memory required for storing the created transposed adjacency list.
+ *             The call expects a valid structure pointed by graph->edgeList and graph->transposedAdjList to be nullptr. Upon successful completion of the call graph->transposedAdjList will point to a gdf_adj_list structure containing the generated transposed adjacency list.
+ *
+ * @Param[out] *graph                cuGRAPH graph descriptor containing the newly added edge list.
```
Comment (afender) [inline cpp/include/functions.h]: param[in, out] as well

File: cpp/include/functions.h [hunk 3]
```diff
@@ -15,65 +15,189 @@
...
- * 
- * @Returns                       GDF_SUCCESS upon successful completion. 
+ *
+ * @Returns                       GDF_SUCCESS upon successful completion.
  */
 /* ----------------------------------------------------------------------------*/
 gdf_error gdf_pagerank(gdf_graph *graph, gdf_column *pagerank, float alpha, float tolerance, int max_iter, bool has_guess);
 
+/**
+ * @Synopsis   Creates source, destination and value columns based on rmat input
```
Comment (afender) [inline cpp/include/functions.h]: based on rmat input -> based on the specified R-MAT model

File: cpp/include/functions.h [hunk 4]
```diff
@@ -15,65 +15,189 @@
...
+ */
+/* ----------------------------------------------------------------------------*/
 gdf_error gdf_grmat_gen (const char* argv, size_t &vertices, size_t &edges, gdf_column* src, gdf_column* dest, gdf_column* val);
 
+/**
+ * @Synopsis   Performs a breadth first search traversal of a graph starting from a node.
+ *
+ * @Param[in] *graph                 cuGRAPH graph descriptor with a valid edgeList or adjList
+ *
+ * @Param[out] *distances            If set to a valid column, this is populated by distance of every vertex is thr graph from the starting node
```
Comment (afender) [inline cpp/include/functions.h]: "this is populated by distance of every vertex is thr graph from the starting node" -> need rework

File: cpp/include/functions.h (hunk 2 above)
Comment (afender) [reply cpp/include/functions.h]: (there are a couple more occurrences in other functions after this one)

File: cpp/include/functions.h [hunk 5]
```diff
@@ -15,65 +15,189 @@
...
+ *                                               --rmat_edgefactor=<edge-factor>
+ *                                               --rmat_edges=<number-edges>
+ *                                               --rmat_a=<factor> --rmat_b=<factor> --rmat_c=<factor>
+ *                                               --rmat_self_loops If this option is supplied, then self loops will be retained
+ *                                               --rmat_undirected If this option is not mentioned, then the graps will be undirected
+ *                                       Optional arguments:
+ *                                       [--device=<device_index>] Set GPU(s) for testing (Default: 0).
+ *                                       [--quiet]                 No output (unless --json is specified).
+ *                                       [--random_seed]           This will enable usage of random seed, else it will use same seed
+ *                                       [--normalized]\n
```
Comment (afender) [inline cpp/include/functions.h]: We should un-document  `--normalized` as this doesn't seem to have any effect.

File: cpp/include/functions.h [hunk 6]
```diff
@@ -15,65 +15,189 @@
...
+ *                                   Indices must be in the range [0, V-1].
+ * @Param[in] *edge_data (optional)  This pointer can be nullptr. If not, this gdf_column of size E (number of edges) contains the weiht for each edge.
+ *                                   The type expected to be floating point.
+ *
+ * @Param[out] *graph                cuGRAPH graph descriptor containing the newly added adjacency list (edge_data is optional).
+ *
+ * @Returns                          GDF_SUCCESS upon successful completion.
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_adj_list_view (gdf_graph *graph,
```
Comment (afender) [inline cpp/include/functions.h]: we could add a sentence saying that “This function does not delete any existing data in the cuGRAPH graph descriptor ”  for all add and view functions

Comment (kaatish) [reply cpp/include/functions.h]: done

**Comment by raydouglass** [comment]
rerun tests

**Pull request merged**